File: Data Structures & Algorithms/subsets/submission-1.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = [[]]

        for num in nums:
            cur_sub = []
            for r in res:
                item = r + [num]
                cur_sub.append(item)
            res += cur_sub
        
        return res

File: Data Structures & Algorithms/majority-element/submission-1.rs
impl Solution {
    pub fn majority_element(nums: Vec<i32>) -> i32 {
        let mut count = HashMap::new();
        let mut res = 0;
        let mut max_count = 0;

        for num in nums {
            *count.entry(num).or_insert(0) += 1;
            if *count.get(&num).unwrap() > max_count {
                res = num;
            }
            max_count = max_count.max(*count.get(&num).unwrap())
        }
        res
    }
}
